evennia/contrib/worldloader/loader.py

```diff
@@ -0,0 +1,491 @@
+        key: (string) Key of the data info.
+    """
+    obj.attributes.add("app", app, category=DATA_INFO_CATEGORY, strattr=True)
+    obj.attributes.add("model", model, category=DATA_INFO_CATEGORY, strattr=True)
+    obj.attributes.add("key", key, category=DATA_INFO_CATEGORY, strattr=True)
+    
+    if (not app) or (not model) or (not key):
+        return True
+
+    return load_data(obj)
+
+
+def load_data(obj):
+    """
+    Load object data from db, and set them to the obj."
+    
+    Args:
+        obj: Object in game.
+    """
+    if not obj:
+        return
+
+    # Get app, model and key names.
+    app = obj.attributes.get(key="app", category=DATA_INFO_CATEGORY, strattr=True)
+    if not app:
+        return False
+
+    model = obj.attributes.get(key="model", category=DATA_INFO_CATEGORY, strattr=True)
+    if not model:
+        return False
+
+    key = obj.attributes.get(key="key", category=DATA_INFO_CATEGORY, strattr=True)
+    if not key:
+        return False
+
+    # Get db model
+    model_obj = get_model(app, model)
+    if not model_obj:
+        logger.log_errmsg("%s can not open model %s" % (key, model))
+        return False
+    
+    # Get data record.
+    data_info = model_obj.objects.filter(key=key)
+    if not data_info:
+        logger.log_errmsg("%s can not find key %s" % (key, key))
+        return False
+
+    info = data_info[0]
+
+    if info.typeclass:
+        set_obj_typeclass(obj, info.typeclass)
+    if info.name:
+        set_obj_name(obj, info.name)
+    if info.alias:
+        set_obj_alias(obj, info.alias)
+    if info.location:
+        set_obj_location(obj, info.location)
+    if info.home:
+        set_obj_home(obj, info.home)
+    if info.desc:
+        set_obj_desc(obj, info.desc)
+    if info.lock:
+        set_obj_lock(obj, info.lock)
+    if info.destination:
+        set_obj_destination(obj, info.destination)
+
+    # Set attributes.
+    attributes = {}
+    if info.attributes:
+        try:
+            # info.attributes: (string) Attribues in form of python dict. Such as: "{'age':'22', 'career':'warrior'}"
+            # Convert string to dict
+            attributes = ast.literal_eval(info.attributes)
+        except Exception, e:
+            logger.log_errmsg("%s can't load attributes %s: %s" % (get_info_key(obj), info.attributes, e))
+
+    # Add other fields to attributes.
+    known_fields = {"key",
+                    "name",
+                    "alias",
+                    "typeclass",
+                    "location",
+                    "home",
+                    "desc",
+                    "lock",
+                    "destination",
+                    "attributes"}
+
+    for field in model_obj._meta.fields:
+        if not field.name in known_fields:
+            attributes[field.name] = info.serializable_value(field.name)
+
+    set_obj_attributes(obj, attributes)
+
+    return True
+
+
+
+def set_obj_typeclass(obj, typeclass):
```

Comment:
You're gonna need to explain this one to me, I'm not quite following. You are here loading data from a custom model and apply it onto the object tied to that data store ... why is this data not applied directly at your `create_object` step?